client/src/EntryForm.js

```diff
@@ -15,12 +15,62 @@ export default function EntryForm({ entry, onSubmit }) {
   function handleSubmit(event) {
     event.preventDefault();
     const newEntry = { title, photoUrl, notes };
-    if (entry) {
-      updateEntry({ ...entry, ...newEntry });
-    } else {
-      addEntry(newEntry);
+    try {
+      if (entry) {
+        addUpdateEntry(entry.entryId, newEntry);
+      } else {
+        addEntry(newEntry);
+      }
+      onSubmit();
+    } catch (err) {
+      console.error(err);
     }
-    onSubmit();
+  }
+
+  async function addEntry(newEntry) {
```

Comment:
I like how you pull these functions outside of the useEffect and event handlers. That makes those functions small and clean.

I probably would have put them in a different file, or at the top level of this file.